mir_planning/mir_pddl_problem_generator/ros/include/mir_pddl_generator_node/pddl_problem_generator.h

```diff
@@ -0,0 +1,67 @@
+
+#ifndef PDDL_PROBLEM_GENERATOR_H
+#define PDDL_PROBLEM_GENERATOR_H
+
+#include <ros/ros.h>
+#include <std_msgs/String.h>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <utility>
+#include <tuple>
+
+#include "rosplan_knowledge_msgs/GetDomainNameService.h"
+#include "rosplan_knowledge_msgs/GetDomainTypeService.h"
+#include "rosplan_knowledge_msgs/GetDomainAttributeService.h"
+#include "rosplan_knowledge_msgs/GetInstanceService.h"
+#include "rosplan_knowledge_msgs/GetAttributeService.h"
+#include "rosplan_knowledge_msgs/GetMetricService.h"
+
+class PDDLProblemGenerator
+{
+    public:
+        PDDLProblemGenerator();
+        ~PDDLProblemGenerator();
+
+        bool generatePDDLProblemFile(std::string problem_path);
+
+    private:
+        ros::NodeHandle nh_;
+
+		std::string domain_name_service_;
+		std::string domain_type_service_;
+		std::string domain_predicate_service_;
+
+		std::string state_instance_service_;
+		std::string state_proposition_service_;
+		std::string state_goal_service_;
+
+        std::map<std::string, float> points_map_;
+        int max_goals_;
+        bool prefer_goals_with_same_source_ws_;
+
+        float getPoints(std::string key);
```

Comment:
it is best practice to use const reference for arguments that will not be changed in the function; i.e. `const std::string &key`. Same for the next four functions below.